crates/subspace-networking/src/utils/piece_provider.rs

```diff
@@ -145,71 +145,45 @@ where
         let key = RecordKey::from(piece_index.to_multihash());
 
         let mut request_batch = self.node.get_requests_batch_handle().await;
-        let get_providers_result = request_batch.get_providers(key.clone()).await;
-
-        match get_providers_result {
-            Ok(mut get_providers_stream) => {
-                while let Some(provider_id) = get_providers_stream.next().await {
-                    trace!(
-                        %piece_index,
-                        key = hex::encode(&key),
-                        %provider_id,
-                        "get_providers returned an item"
-                    );
+        let mut get_providers_stream = request_batch
+            .get_providers(key.clone())
+            .await
+            .inspect_err(|err| warn!(%piece_index,?key, ?err, "get_providers returned an error"))
+            .ok()?;
+
+        let key = hex::encode(&key);
```

Comment:
Just a note for future reference: previously, this hex encoding call was inside tracing macros, so it wouldn't run unless the messages were actually logged.

In this specific case, encoding 128 bytes is cheap, so it is unlikely to impact performance.